lib/matplotlib/cbook.py

```diff
@@ -2445,6 +2445,14 @@ def safe_first_element(obj):
     return next(iter(obj))
 
 
+def asSequence(data):
```

Comment:
This looks tricky to get right.  It seems contrary to the spirit of duck-typing we have historically favored.  Objects can be array-like, and work fine in mpl, without being subclasses of anything we know about.  I don't think we want to start issuing warnings in such cases.  Nor can we do all of our usual homogenization at this very early stage; we have to preserve those pesky units, and decisions about missing values need to be handled by each function on an argument by argument basis.